Sources/SwiftSyntax/SyntaxNodes.swift.gyb.template

```diff
@@ -70,14 +70,28 @@ public struct ${node.name}: ${base_type}Protocol, SyntaxHashable {
   public init?(_ syntax: Syntax) {
     guard syntax.raw.kind == .${node.swift_syntax_kind} else { return nil }
     self._syntaxNode = syntax
+%   if node.must_uphold_invariant:
+    if !isValid {
+      return nil
+    }
+%   end
```

Comment:
I think it’s save to make it a `fatalError` if the node is not valid at this stage. By our invariant, we know that every `Syntax` node is valid and casting it should not change the invariant.

Reply:
Done.